port_ocean/clients/port/mixins/integrations.py

```diff
@@ -50,13 +51,36 @@ async def get_log_attributes(self) -> LogAttributes:
             self._log_attributes = response["logAttributes"]
         return self._log_attributes
 
+    async def _poll_integration_until_default_provisioning_is_complete(
+        self,
+        interval=15,
+    ) -> Dict[str, Any]:
+        response = await self._get_current_integration()
+        response_json = response.json()
+        config = response_json.get("integration", {}).get("config", {})
+        if config != {}:
+            return response_json
+
+        logger.info(
+            f"Still waiting for integration config to be ready, waiting {interval} seconds"
+        )
+        await asyncio.sleep(interval)
+
+        # TODO: Ensure that get_integration isn't cached
+        result = await self._poll_integration_until_default_provisioning_is_complete()
+
+        return result
```

Comment:
why is this recursion in needed?

Reply:
that's the polling until the config is ready

Reply:
Removed recursion, added backoff mechanism